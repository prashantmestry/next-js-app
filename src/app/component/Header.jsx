import Image from "next/image";
import Link from "next/link";

const Header = () => {
  return (
    <header className="row-start-3 flex gap-6 flex-wrap items-center justify-center">
      <Link href="/about">About</Link>
    </header>
  );
};

export default Header;
