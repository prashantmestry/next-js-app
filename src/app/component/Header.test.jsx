import React from "react";
import "@testing-library/jest-dom";
import { renderer } from "react-test-renderer";

import { render, screen } from "@testing-library/react";
import Header from "./Header";

describe("Header component testing", () => {
  test("Title check", () => {
    render(<Header />);
    const title = screen.getByText(/about/i);
    expect(title).toBeInTheDocument();
  });

  test("Header snapshot test", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
