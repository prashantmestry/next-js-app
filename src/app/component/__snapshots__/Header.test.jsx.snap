// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component testing Header snapshot test 1`] = `
<div>
  <header
    class="row-start-3 flex gap-6 flex-wrap items-center justify-center"
  >
    <a
      href="/about"
    >
      About
    </a>
     work
  </header>
</div>
`;
